### DISCUSSION TITLE
Cannot find module '@sequelize/core/decorators-legacy' or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by bcetienne** [sequelize/sequelize#16425] [Help & Questions] [answered]
Hi everyone.

I was working on a project with Apollo and I keep all my packages up to date.

So I switched to TypeScript 5.2 except that Sequelize V6 doesn't seem fully compatible, and I was getting errors when compiling the code.

Looking at this Sequelize [GitHub issue](https://github.com/sequelize/sequelize/issues/16414), it says this is fixed with Sequelize version 7.

Seeing that there's a lot of change with version 6 (especially class with decorators instead of `.init`), I decided I wanted to start getting a head start and migrate to this new version, except I'm running into a problem I can't solve.

I installed `@sequelize/core` as specified, and imported the packages for the legacy decorators.

https://sequelize.org/api/v7/modules/_sequelize_core.html
https://sequelize.org/api/v7/modules/_sequelize_core.decorators_legacy

However, any IDE and even the compiler tell me they can't find the @sequelize/core/decorators-legacy module.

Has anyone ever found themselves in this situation?

Here is a snap of my `package.json`

```json
{
  "dependencies": {
    ...,
    "@sequelize/core": "^7.0.0-alpha.27",
    ...
  },
  "devDependencies": {
    ...,
    "sequelize-cli": "^6.6.1",
    "typescript": "5.2.2"
    ...
  }
}
```

And here one of my model

```typescript
import {
  Model,
  InferAttributes,
  InferCreationAttributes,
  CreationOptional,
  DataTypes,
} from "@sequelize/core";
import {
  Attribute,
  PrimaryKey,
  AutoIncrement,
  NotNull,
  Index,
} from "@sequelize/core/decorators-legacy";

export class Visibility extends Model<
  InferAttributes<Visibility>,
  InferCreationAttributes<Visibility>
> {
  @Attribute(DataTypes.INTEGER)
  @PrimaryKey
  @AutoIncrement
  declare id: CreationOptional<number>;

  @Attribute(DataTypes.STRING)
  @NotNull
  @Index({ unique: true })
  declare name: string;

  @Attribute(DataTypes.TEXT)
  declare description: string;
}
```

Thanks if someone has a solution

**Comment by xmtg2o7** [comment]
I was also having this issue; while this may not help in your case, I ended up using _require_ instead of _import_ and it worked for me.

`const { Attribute, PrimaryKey } = require('@sequelize/core/decorators-legacy');`

**Comment by ephys** [comment, marked as answer]
This is likely the problem you're experiencing: https://github.com/sequelize/website/issues/552, see the comments for the solution

**Reply by xmtg2o7** [reply]
@ephys 

I appreciate the input, while setting _moduleResolution_ to either 'Node16' or 'NodeNext' did not work, changing just the _module_ to 'Node16' did. I can now do an import without issue.
